src/entry.sh

```diff
@@ -1,562 +1,744 @@
-echo "fhem ALL=(ALL) NOPASSWD: /usr/bin/apt-get -y -q -V dist-upgrade" >> /etc/sudoers.d/fhem-docker
-echo "fhem ALL=(ALL) NOPASSWD:SETENV: /usr/bin/npm update *" >> /etc/sudoers.d/fhem-docker
+#====================================================================================================================-
+#--- FHEM utility functions ------------------------------------------------------------------------------------------
 
-# Allow installation of new packages
-echo "fhem ALL=(ALL) NOPASSWD:SETENV: /usr/local/bin/cpanm *" >> /etc/sudoers.d/fhem-docker
-echo "fhem ALL=(ALL) NOPASSWD: /usr/bin/apt-get -y install *" >> /etc/sudoers.d/fhem-docker
-echo "fhem ALL=(ALL) NOPASSWD:SETENV: /usr/bin/npm install *" >> /etc/sudoers.d/fhem-docker
-echo "fhem ALL=(ALL) NOPASSWD:SETENV: /usr/bin/npm uninstall *" >> /etc/sudoers.d/fhem-docker
-
-chmod 440 /etc/sudoers.d/fhem*
-chown --quiet --no-dereference root:${FHEM_GID} /etc/sudoers.d/fhem* 2>&1>/dev/null
-(( i++ ))
-
-# SSH key: Ed25519
-mkdir -p ${FHEM_DIR}/.ssh
-chmod 700 ${FHEM_DIR}/.ssh
-[ -e ${FHEM_DIR}/.ssh/authorized_keys ] && chmod 600 ${FHEM_DIR}/.ssh/authorized_keys
-if [ ! -s ${FHEM_DIR}/.ssh/id_ed25519 ]; then
-  echo "$i. Generating SSH Ed25519 client certificate for user 'fhem' ..."
-  rm -f ${FHEM_DIR}/.ssh/id_ed25519*
-  ssh-keygen -t ed25519 -f ${FHEM_DIR}/.ssh/id_ed25519 -q -N "" -o -a 100
-  sed -i "s/root@.*/fhem@fhem-docker/" ${FHEM_DIR}/.ssh/id_ed25519.pub
-  (( i++ ))
-fi
+# Prepend a relative path with the FHEM_DIR.
+# Works correctly still when the given path is already prepended by FHEM_DIR
+#
+# Usage: prependFhemDirPath path
+# Parameters:  path         Path to prepend
+# Global vars: FHEM_DIR
+#
+function prependFhemDirPath() {
+  realpath -ms "$(echo "$1" | sed -e 's!^'${FHEM_DIR}/'!!; s!^!'${FHEM_DIR}/'!')"
+}
 
-# SSH key: RSA
-if [ ! -s ${FHEM_DIR}/.ssh/id_rsa ]; then
-  echo "$i. Generating SSH RSA client certificate for user 'fhem' ..."
-  rm -f ${FHEM_DIR}/.ssh/id_rsa*
-  ssh-keygen -t rsa -b 4096 -f ${FHEM_DIR}/.ssh/id_rsa -q -N "" -o -a 100
-  sed -i "s/root@.*/fhem@fhem-docker/" ${FHEM_DIR}/.ssh/id_rsa.pub
-  (( i++ ))
-fi
 
-# SSH client hardening
-if [ ! -f ${FHEM_DIR}/.ssh/config ]; then
-  echo "$i. Generating SSH client configuration for user 'fhem' ..."
-echo "IdentityFile ~/.ssh/id_ed25519
-IdentityFile ~/.ssh/id_rsa
+# Get the value of a global attribute from the FHEM config file.
+#
+# Usage: getGlobalAttr file attribute
+# Parameters:  file         File FHEM config file
+#              attribute    Attribute to look for
+# Returns:     0:     Attribute found and value sent to stdout
+#              else:  An error occured, e.g. attribute not in file
+#
+function getGlobalAttr() {
+  local -r inCfgFile="$1"
+  local -r inAttr="$2"
+  awk 'BEGIN{ retVal=1} /^[[:space:]]*attr[[:space:]]+global[[:space:]]+'$inAttr'[[:space:]]+/{ print $4; retVal=0; } END{ exit retVal}' "$inCfgFile"
```

Comment:
I think here is a bug. If i add a test which adds attribute logfile to the config file it is not detected 

```
    echo "attr global logfile ./log/fhem-%Y-%m.log" >> /opt/fhem/fhem.cfg
    assert_file_contains /opt/fhem/fhem.cfg "attr global logfile"
    run -0 getGlobalAttr /opt/fhem/fhem.cfg "logfile"
 ```

Reply:
I've fixed this not using awk, because mawk is installed in the image.